### PR TITLE
Merging user creation and selection testing branch

### PR DESCRIPTION
**Pull request by jokubassilgalys**

### CODE REVIEW

**Review by ScottasM — APPROVED** [review]

File: PPI-projektas/Controllers/UserController.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            var userService = new UserService();
+            if (!userService.ValidateData(userData)) return BadRequest("Invalid Data");
+
+            var userGuidId = userService.CreateUser(userData);
+
+            return CreatedAtAction("CreateUser", userGuidId);
+        }
+
+        [HttpDelete("delete/{userId:guid}")]
+        public IActionResult Delete(Guid id)
```
Comment (titasstank) [inline PPI-projektas/Controllers/UserController.cs:36]: mismatch between route and method argument

File: PPI-projektas/Services/UserServices.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+using PPI_projektas.Utils;
+
+namespace PPI_projektas.Services;
+
+public class UserService
+{
+    public bool ValidateData(string data)
+    {
+        if (String.IsNullOrEmpty(data)) return false;
+        else return true;
```
Comment (titasstank) [inline PPI-projektas/Services/UserServices.cs:12]: can be rewritten to just "return !String.IsNullOrEmpty(data)"

File: PPI-projektas/Services/UserServices.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+    public bool ValidateData<T>(List<T> data)
+    {
+        if (data == null) return false;
+        else return true;
+    }
+    public bool ValidateData(UserCreateData data)
+    {
+        if(data.Equals(default(UserCreateData))) return false;
+        else return true;
```
Comment (titasstank) [inline PPI-projektas/Services/UserServices.cs:22]: can be rewritten to just "return !data.Equals(default(UserCreateData))"

File: PPI-projektas/objects/User.cs [hunk 4]
```diff
@@ -1,41 +1,61 @@
...
     
-    [JsonIgnore] public List<Group> Groups;
+    [JsonIgnore] public List<Group> Groups = new();
     [JsonInclude] public List<Guid> GroupsGuids;
 
     public User () {} // For deserialization
-    
-    public User(string name, string password, string email)
+
+    public User(string name, string password, string email, bool createGUID = true) : base(createGUID)
```
Comment (titasstank) [inline PPI-projektas/objects/User.cs:25]: use : this() and add only missing fields

**Review by titasstank — CHANGES_REQUESTED** [review]

**Review by titasstank — APPROVED** [review]

**Review by ZygimantasVidmantas — APPROVED** [review]

**Pull request merged**